service/src/main/kotlin/fi/espoo/evaka/vasu/DocumentMigrator.kt

```diff
@@ -0,0 +1,550 @@
+    templateContent: DocumentTemplateContent
+): DocumentTemplateId {
+    tx.createUpdate { sql("LOCK TABLE document_template") }.execute()
+    return getMatchingTemplate(tx, templateBasics, templateContent)
+        ?: tx.insertTemplate(templateBasics)
+            .id
+            .also { tx.updateDraftTemplateContent(it, templateContent) }
+            .also { tx.publishTemplate(it) }
+}
+
+private fun getMatchingTemplate(
+    tx: Database.Transaction,
+    templateBasics: DocumentTemplateBasicsRequest,
+    templateContent: DocumentTemplateContent
+): DocumentTemplateId? =
+    tx.createQuery {
+            sql(
+                """
+    SELECT id FROM document_template
+    WHERE name = ${bind(templateBasics.name)}
+     AND type = ${bind(templateBasics.type)}
+     AND language = ${bind(templateBasics.language)}
+     AND confidential = ${bind(templateBasics.confidential)}
+     AND legal_basis = ${bind(templateBasics.legalBasis)}
+     AND validity = ${bind(templateBasics.validity)}
+     AND content = ${bind(templateContent)}::jsonb
+"""
+            )
+        }
+        .exactlyOneOrNull()
+
+private fun Database.Transaction.deletePreviouslyMigratedChildDocument(id: ChildDocumentId) {
+    createUpdate {
+            sql(
+                """
+        DELETE FROM child_document cd WHERE id = ${bind(id)} AND EXISTS(
+            SELECT 1 FROM document_template dt
+            WHERE cd.template_id = dt.id AND dt.type IN ('MIGRATED_VASU', 'MIGRATED_LEOPS')
+        )
+    """
+            )
+        }
+        .execute()
+}
+
+private fun Database.Transaction.insertMigratedChildDocument(
+    id: ChildDocumentId,
+    childId: ChildId,
+    templateId: DocumentTemplateId,
+    content: DocumentContent,
+    modifiedAt: HelsinkiDateTime,
+    publishedAt: HelsinkiDateTime
+) {
+    createUpdate {
+            sql(
+                """
+INSERT INTO child_document(id, child_id, template_id, status, content, published_content, modified_at, published_at)
+VALUES (${bind(id)}, ${bind(childId)}, ${bind(templateId)}, 'COMPLETED', ${bind(content)}, ${bind(content)}, ${bind(modifiedAt)}, ${bind(publishedAt)})
+"""
+            )
+        }
+        .execute()
+}
+
+private fun toBasicsRequest(vasu: VasuDocument): DocumentTemplateBasicsRequest {
+    val translations = getTranslations(vasu.language)
+    return DocumentTemplateBasicsRequest(
+        name = vasu.templateName,
+        type =
+            when (vasu.type) {
+                CurriculumType.DAYCARE -> DocumentType.MIGRATED_VASU
+                CurriculumType.PRESCHOOL -> DocumentType.MIGRATED_LEOPS
+            },
+        language =
+            when (vasu.language) {
+                VasuLanguage.FI -> DocumentLanguage.FI
+                VasuLanguage.SV -> DocumentLanguage.SV
+            },
+        confidential = true,
+        legalBasis =
+            when (vasu.type) {
+                CurriculumType.DAYCARE ->
+                    when (vasu.language) {
```

Comment:
Looks like you are always returning translations.lawVasu so you don't need to check the language here
```suggestion
                    translations.lawVasu
```

Reply:
Ah yeah, that was from time when translations were hard coded here, cleaned it up now